eng/native/configureplatform.cmake

```diff
@@ -91,6 +91,7 @@ endif(CLR_CMAKE_HOST_OS STREQUAL Darwin)
 if(CLR_CMAKE_HOST_OS STREQUAL iOS)
     set(CLR_CMAKE_HOST_UNIX 1)
     set(CLR_CMAKE_HOST_IOS 1)
+    set(CLR_CMAKE_HOST_DARWIN 1)
```

Comment:
@am11 this is wrong, the way `Darwin` is used in CMake means that it really means OSX/macOS. We should revert this change and rename `CLR_CMAKE_HOST_DARWIN` to `CLR_CMAKE_HOST_OSX`

Reply:
@akoeplinger, in libraries, there are only few usages of Darwin in CMake, and I have kept it generalized (unless specialization is really need (i.e. one place), same way we do for linux, win32), and also made sure the builds are green.

Reply:
In my experience most of the time the logic is really OSX-specific so having a common symbol doesn't make a lot of sense and prevents errors like https://github.com/dotnet/runtime/blob/ff8e91121ced72b72daddee2a4c9b16679223fb4/src/libraries/Native/Unix/configure.cmake#L13-L14

I filed https://github.com/dotnet/runtime/pull/33959 to make that clearer and there are only three or four places where a common symbol would be used.